### PR TITLE
refactor: deprecate util.find_mercurial_ancestor

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/nvim-lspconfig/issues/2079.

### CODE REVIEW

**Comment by dundargoc** [comment]
Dependent on https://github.com/neovim/nvim-lspconfig/pull/3475

File: lua/lspconfig/util.lua [hunk 1]
```diff
@@ -405,4 +396,9 @@ function M.path.exists(filename)
   return stat and stat.type or false
 end
 
+--- @deprecated use `vim.fs.find('.hg', { path = startpath, upward = true })` instead
+function M.find_mercurial_ancestor(startpath)
+  return vim.fs.find('.hg', { path = startpath, upward = true })
```
Comment (glepnir) [inline lua/lspconfig/util.lua:401]: A limit should be set here. And it is possible that multiple matching directories will be returned. The first one found may not be the correct root directory

Comment (dundargoc) [reply lua/lspconfig/util.lua:401]: Wait what? How does one identify a "real" root directory? I assumed it meant the first parent directory containing ".hg"?

Comment (glepnir) [reply lua/lspconfig/util.lua:401]: Oh.. I misread this. I was confused and thought it was node search. Sorry I haven't slept for 35 hours so this is a bit confusing. nvm

**Review by glepnir — APPROVED** [review]

**Pull request merged**